pages/services/confluent-kafka/2.4.0-4.1.1/api-reference/index.md

```diff
@@ -0,0 +1,271 @@
+[
+  "topic1",
+  "topic0"
+]
+```
+
+```bash
+$ curl -H "Authorization: token=$auth_token" "<dcos_url>/service/{{ model.serviceName }}/v1/topics"
+[
+  "topic1",
+  "topic0"
+]
+```
+
+## Describe Topic
+
+```bash
+$ dcos {{ model.packageName }} --name={{ model.serviceName }} topic describe topic1
+{
+  "partitions": [
+  {
+    "0": {
+      "controller_epoch": 1,
+      "isr": [
+        0,
+        1,
+        2
+      ],
+      "leader": 0,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  },
+  {
+    "1": {
+      "controller_epoch": 1,
+      "isr": [
+        1,
+        2,
+        0
+      ],
+      "leader": 1,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  },
+  {
+    "2": {
+      "controller_epoch": 1,
+      "isr": [
+        2,
+        0,
+        1
+      ],
+      "leader": 2,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  }
+  ]
+}
+```
+
+```bash
+$ curl -X POST -H "Authorization: token=$auth_token" "<dcos_url>/service/{{ model.serviceName }}/v1/topics/topic1"
+{
+  "partitions": [
+  {
+    "0": {
+      "controller_epoch": 1,
+      "isr": [
+        0,
+        1,
+        2
+      ],
+      "leader": 0,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  },
+  {
+    "1": {
+      "controller_epoch": 1,
+      "isr": [
+        1,
+        2,
+        0
+      ],
+      "leader": 1,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  },
+  {
+    "2": {
+      "controller_epoch": 1,
+      "isr": [
+        2,
+        0,
+        1
+      ],
+      "leader": 2,
+      "leader_epoch": 0,
+      "version": 1
+    }
+  }
+  ]
+}
+
+```
+
+## Create Topic
+
+```bash
+$ dcos {{ model.packageName }} --name={{ model.serviceName }} topic create topic1 --partitions=3 --replication=3
+{
+  "message": "Output: Created topic \"topic1\"\n"
+}
+```
+
+```bash
+$ curl -X POST -H "Authorization: token=$auth_token" "<dcos_url>/service/{{ model.serviceName }}/v1/topics/topic1?partitions=3&replication=3"
+{
+  "message": "Output: Created topic \"topic1\"\n"
+}
+```
+
+## View Topic Offsets
+
+There is an optional `--time` parameter which may be set to either "first", "last", or a timestamp in milliseconds as [described in the Kafka documentation][15].
+
+```bash
+$ dcos {{ model.packageName }} --name={{ model.serviceName }} topic offsets topic1 --time=last
+[
+  {
+    "2": "334"
+  },
+  {
+    "1": "333"
+  },
+  {
+    "0": "333"
+  }
+]
+```
+
+```bash
+$ curl -H "Authorization: token=$auth_token" "<dcos_url>/service/{{ model.serviceName }}/v1/topics/topic1/offsets?time=-1"
+[
+  {
+    "2": "334"
+  },
+  {
+    "1": "333"
+  },
+  {
+    "0": "333"
+  }
+]
+```
+
+## Alter Topic Partition Count
+
+```
+$ dcos {{ model.packageName }} --name={{ model.serviceName }} topic partitions topic1 2
+{
+  "message": "Output: WARNING: If partitions are increased for a topic that has a key, the partition logic or ordering of the messages will be affectednAdding partitions succeeded!n"
```

Comment:
This message doesn't look right. What does "affectednAdding partitions succeeded!n" mean? Is this correct?

Reply:
The `n` stuck in the middle between `addected` and `Adding` should be `\n` indicating a newline. The same holds for the final `n`.

Reply:
Will fix.